app/models/user.rb

```diff
@@ -320,17 +320,21 @@ def reindex_referencing_models
     return if no_xapian_reindex == true
     return unless saved_change_to_attribute?(:url_name)
 
+    expire_comments
+    expire_requests
+  end
+
+  def expire_requests
+    info_requests.find_each(&:expire)
```

Comment:
`InfoRequest#expire` will clear fragment and database cache. Are you sure it is okay to do this every time the user is updated? Seems like this could cause a lot of extra processing.

Reply:
Sorry, clearly I should have been more explicit about this change in the commit message.

We [decided to do this](https://github.com/mysociety/alaveteli/pull/6528#discussion_r716576957) for the analogous `PublicBody` method. This will only get called when there's been a change to `url_name`, and given we don't rename users often I don't think it's much of an issue.

I agree that all this cache expiry stuff isn't great, but my aim here has been to at least reduce the complexity – even at the expense of a little extra processing – in order to hopefully extract a common request expiry mechanism down the line.

Reply:
Ah yes. I missed https://github.com/mysociety/alaveteli/pull/6644/files#diff-9802ca3c9c4cf89904fd44bc114e35ebdf2c5dd3d5b645491e2b253e1afef29bR321

I think this is fine to merge then